### PR TITLE
Add verrazzano_component label to kubelet and kube-state-metrics metrics

### PR DESCRIPTION
**Pull request by lake-of-dreams**
The regex for each component for pod level metrics is added in pattern
- `"(<namespace>);(<componentName>);(<regex-pattern-for-pod>)(.*)"`
and for pvc level metrics
- `"(<namespace>);(<componentName>);(<regex-pattern-for-pvc>)(.*)"
so that capture group `$2` will give target value of `verrazzano_component`

Also added an e2e test to check health status of `kubelet` and `kube-state-metrics` endpoints

### CODE REVIEW

File: platform-operator/controllers/verrazzano/component/common/prometheus.go [hunk 1]
```diff
@@ -19,7 +19,60 @@ import (
...
+const (
+	keycloakHTTPService = "keycloak-http"
+
+	// LabelVerrazzanoComponent is name of label containing Verrazzano component name for a metrics.
+	LabelVerrazzanoComponent = "verrazzano_component"
+)
+
+// verrazzanoComponentRelabelConfigPodTemplates is mapping of Verrazzano component names to the regular expression template for capturing pod related metrics.
+var verrazzanoComponentRelabelConfigPodTemplates = map[string]string{
+	"Alertmanager":                    "(verrazzano-monitoring);(%s);(alertmanager-[a-z0-9-]+)(.*)",
```
Comment (desagar) [inline platform-operator/controllers/verrazzano/component/common/prometheus.go:31]: The label values seem named inconsistently - there’s “Verrazzano Monitoring Operator”, “Alertmanager”, “cert-manager” - I’m thinking we should align these names to the component names in the VZ CR.

**Review by fredtibbitts — COMMENTED** [review]
The metrics relabelings can be done right in the helm chart with helm overrides. The kube-state-metrics service monitor already supports this. See platform-operator/helm_config/overrides/kube-state-metrics-values.yaml.

It seems like it would be much cleaner and simpler to put the relabeling config in the helm overrides.

File: platform-operator/controllers/verrazzano/component/common/prometheus.go (hunk 1 above)
Comment (lake-of-dreams) [reply platform-operator/controllers/verrazzano/component/common/prometheus.go]: fixed

**Comment by lake-of-dreams** [comment]
> The metrics relabelings can be done right in the helm chart with helm overrides. The kube-state-metrics service monitor already supports this. See platform-operator/helm_config/overrides/kube-state-metrics-values.yaml.
> 
> It seems like it would be much cleaner and simpler to put the relabeling config in the helm overrides.

moved relabelling to overrides

**Pull request merged**